src/request.rs

```diff
@@ -33,6 +34,21 @@ impl Request {
     }
 }
 
+impl<T> From<HttpRequest<T>> for Request where T: Into<Body> {
+    fn from(r: HttpRequest<T>) -> Self {
+        let header = RequestHeader {
+            url: Url::parse(&format!("{}", r.uri())).unwrap(),
```

Comment:
I will merge your PR but I'll add a note here that this should be better handled in the future. (Invalid urls could be used as an exploit to make a software panic, but it's not like the crate is security oriented to begin with.)

Also see: https://github.com/seanmonstar/reqwest/issues/668

Reply:
That issue is about the other direction, but I'd not be surprised if this way also had parsing disagreements. Unfortunately, there doesn't seem any work going on to make a unified `url` crate that `http` is willing to use (builder pattern in use, HTTP2 breakdown and relative path support), so we're stuck with these conversion warts.

hyperium/http#396 hyperium/http#277 (points to servo/rust-url#468) hyperium/http#137 hyperium/http#73